mrp_subcontracting_stock_owner_restriction/models/mrp_production.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    def write(self, vals):
+        if "owner_id" in vals:
+            for production in self:
+                if production.owner_restriction in (
+                    "unassigned_owner",
+                    "picking_partner",
```

Comment:
I think it makes sense to include `partner_or_unassigned` as well.
```suggestion
                    "picking_partner",
                    "partner_or_unassigned",
```